mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksViewModel.kt

```diff
@@ -236,25 +244,34 @@ abstract class MavericksViewModel<S : MavericksState>(
         }
 
         setState { reducer(Loading()) }
-        return onEach {
-            setState { reducer(Success(it)) }
-        }.launchIn(viewModelScope + dispatcher)
+
+        return catch { error -> setState { reducer(Fail(error)) } }
+            .onEach {
+                setState { reducer(Success(it)) }
+            }.launchIn(viewModelScope + (dispatcher ?: EmptyCoroutineContext))
```

Comment:
can these two lines be consistent:
```suggestion
        return catch { setState { reducer(Fail(it)) } }
            .onEach { setState { reducer(Success(it)) } }
            .launchIn(viewModelScope + (dispatcher ?: EmptyCoroutineContext))
```

or 
```suggestion
        return catch { error -> setState { reducer(Fail(error)) } }
            .onEach { value -> setState { reducer(Success(value)) } }
            .launchIn(viewModelScope + (dispatcher ?: EmptyCoroutineContext))
```